### PR TITLE
Basic handling of reorgs

### PR DESCRIPTION
**Pull request by darosior**
This turned out to have even more edge cases than i expected. Went for a bulk (rescan everything on reorg) method instead of a more smarter-and-footguny (keep a header chain and rescan based on deposit blockheight) one. See commit messages for more comprehensive details.

Note that this is only a partial implementation, as we'll eventually have to deal with all the remaining transactions.

Given the slippery slop that it is by itself i separated the functional test in a previous commit for reviewers to be able to evaluate the edge cases coverage. A slippery slop it's even more with the concurrency (RPC thread, signature fetcher) but for those ones the race is so intense it can't be tested reproducibly.

### CODE REVIEW

File: tests/serializations.py [hunk 1]
```diff
@@ -25,15 +25,19 @@
 import copy
```
Comment (edouardparis) [inline tests/serializations.py:25]: looks like Blake did a big job

Comment (darosior) [reply tests/serializations.py:25]: Glad you like it, it's increasing the diff and the blame depth but worth it imho

**Review by edouardparis — APPROVED** [review]
ACK d16c90a1a35bdd1bdb294c30af1f342697c2ed55
